pkg/credentials/sts_tls_identity.go

```diff
@@ -145,7 +115,19 @@ func (i *STSCertificateIdentity) Retrieve() (Value, error) {
 	}
 	req.Form.Add("DurationSeconds", strconv.FormatUint(uint64(livetime.Seconds()), 10))
 
-	resp, err := i.Client.Do(req)
+	tr, ok := cc.Client.Transport.(*http.Transport)
+	if !ok {
+		return Value{}, fmt.Errorf("CredContext should contain an http.Transport value")
+	}
+
+	trCopy := tr.Clone()
+	trCopy.TLSClientConfig.Certificates = []tls.Certificate{i.Certificate}
```

Comment:
This was the best alternative that I could come up with. We don't want to change the `TLSClientConfig` of the original HTTP transport, because that may have security and parallelization issues.